### PR TITLE
Simplify sync.js

### PR DESCRIPTION
**Pull request by JuhQ**
Simplified and cleaned sync.js

### CODE REVIEW

**Comment by alisd23** [comment]
Tests are failing. Make sure these changes haven't broken the `this` context in some cases.
In general I'm OK with the changes. However as there's are such minor changes can you make doubly sure this has not affected the functionality? Thanks

File: src/sync.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
 
 export const syncHistoryWithStore = (history, store) => {
   // Initialise store
   store.history = history;
 
   // Handle update from history object
-  const handleLocationChange = (location) => {
-    store._updateLocation(location);
-  };
+  const handleLocationChange = store._updateLocation;
```
Comment (alisd23) [inline src/sync.js]: This change will break the `_updateLocation` call, as the `this` context will not be bound correctly. Might just be easier to revert this change.

File: src/sync.js [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
   const unsubscribeFromHistory = history.listen(handleLocationChange);
   handleLocationChange(history.location);
 
-  const subscribe = (listener) => {
-    const onStoreChange = (change) => {
-      const rawLocation = { ...store.location };
-      listener(rawLocation, history.action);
+  const subscribe = listener => {
+    const onStoreChange = () => {
+      listener({ ...store.location }, history.action);
```
Comment (alisd23) [inline src/sync.js]: I think I prefer the intermediate variable personally, and the negative effects are entirely negligible.

**Comment by JuhQ** [comment]
@alisd23 changes made based on feedback

**Comment by alisd23** [comment]
Commit messages still don't adhere to angular semantic commit message guidelines, so for the sake of time, I have copied some of the changes over to a new PR: #50.
Thanks for PR though 👍
